integration/spark2/src/main/scala/org/apache/spark/sql/CarbonDataFrameWriter.scala

```diff
@@ -167,13 +167,19 @@ class CarbonDataFrameWriter(sqlContext: SQLContext, val dataFrame: DataFrame) {
     val carbonSchema = schema.map { field =>
       s"${ field.name } ${ convertToCarbonType(field.dataType) }"
     }
+  val isStreaming = if (options.isStreaming) Some("true") else None
+
     val property = Map(
       "SORT_COLUMNS" -> options.sortColumns,
       "DICTIONARY_INCLUDE" -> options.dictionaryInclude,
       "DICTIONARY_EXCLUDE" -> options.dictionaryExclude,
-      "TABLE_BLOCKSIZE" -> options.tableBlockSize
-    ).filter(_._2.isDefined).map(p => s"'${p._1}' = '${p._2.get}'").mkString(",")
+      "TABLE_BLOCKSIZE" -> options.tableBlockSize,
+      "STREAMING" -> isStreaming
```

Comment:
why not use `options.isStreaming` directly?

Reply:
@jackylk reason that i not used options.isStreaming is because it is giving back a boolean value so i just converted the boolean to option

Reply:
It is ok to add `options.isStreaming` directly, by default it is false if user does not specify it.

Reply:
using it directly will cause this test case to fail

test("test datasource table with specified table path") {
val path = "./source"
df2.write
.format("carbondata")
.option("tableName", "carbon10")
.option("tablePath", path)
.mode(SaveMode.Overwrite)
.save()
assert(new File(path).exists())
checkAnswer(
sql("select count() from carbon10 where c3 > 500"), Row(500)
)
sql("drop table carbon10")
assert(!new File(path).exists())
assert(intercept[AnalysisException](
sql("select count() from carbon10 where c3 > 500"))
.message
.contains("not found"))
}

this is because of some problem at parsing level giving tableproperties followed by table path is not parsing correctlly a seperate jira is already created for it here is the link

https://issues.apache.org/jira/browse/CARBONDATA-2005

are we going to support it?

thats why what i do is that if is streaming is false then do not include it in property map make it none

Reply:
Yes, "CREATE TABLE LOCATION" is external table feature, it is supported in #1749, but currently that is not planned to merge into carbonstore branch but not master.

Reply:
I still suspect it should work, did you try 
```
"STREAMING" -> options.isStreaming.toString
```

Reply:
@jackylk  yes i do test case fail with similiar exception as in jira https://issues.apache.org/jira/browse/CARBONDATA-2005

this test case is failing

test("test datasource table with specified table path") {
val path = "./source"
df2.write
.format("carbondata")
.option("tableName", "carbon10")
.option("tablePath", path)
.mode(SaveMode.Overwrite)
.save()
assert(new File(path).exists())
checkAnswer(
sql("select count() from carbon10 where c3 > 500"), Row(500)
)
sql("drop table carbon10")
assert(!new File(path).exists())
assert(intercept[AnalysisException](
sql("select count() from carbon10 where c3 > 500"))
.message
.contains("not found"))
}

Reply:
thats why my my bulid failed at first time 

@jackylk  you can try this in carbonsession example here it is 
Exception in thread "main" org.apache.spark.sql.AnalysisException: == Parse1 ==

mismatched input 'location' expecting <EOF>(line 16, pos 79)

== SQL ==

 CREATE TABLE carbon_table(
 shortField SHORT,
 intField INT,
 bigintField LONG,
 doubleField DOUBLE,
 stringField STRING,
 timestampField TIMESTAMP,
 decimalField DECIMAL(18,2),
 dateField DATE,
 charField CHAR(5),
 floatField FLOAT,
 complexData ARRAY<STRING>
 )
 STORED BY 'carbondata'
 TBLPROPERTIES('SORT_COLUMNS'='', 'DICTIONARY_INCLUDE'='dateField, charField') location '/tmp'
-------------------------------------------------------------------------------^^^

Reply:
@jackylk  i found the root cause ddl written in carbondataframe writer is wrong location should come above of tblproperties the same way it comes in hive so ddl in data framewriter should be

  s"""
       | CREATE TABLE IF NOT EXISTS $dbName.${options.tableName}
       | (${ carbonSchema.mkString(", ") })
       | STORED BY 'carbondata'
       | ${ if (options.tablePath.nonEmpty) s"LOCATION '${options.tablePath.get}'" else ""}
       |  ${ if (property.nonEmpty) "TBLPROPERTIES (" + property + ")" else "" }
       |
     """.stripMargin

i corrected it now and every thing is running fine please review